iterate on toasts

pierrebeitz opened the pull request:
because i wanted to use toasts in DCOS-UI and was interested in how stuff works, i played around with some code and thought it's worthwhile to share the results.

brandonc commented on packages/toaster/components/Toast.tsx:
```diff
@@ -26,21 +26,21 @@ export interface ToastProps {
   description?: React.ReactElement<HTMLElement> | string;
   id: ToastId;
-  appearance?: "default" | "success" | "warning" | "danger";
+  appearance: "default" | "success" | "warning" | "danger";
```
~Is this a breaking change?~

I see the default now

GeorgiSTodorov commented on packages/table/stories/Table.stories.tsx:
```diff
@@ -467,37 +467,37 @@ storiesOf("Table", module)
       <Table data={items}>
         <Column
-          header={<HeaderCell lowerCase={true}>name</HeaderCell>}
+          header={<HeaderCell capitalize={false}>name</HeaderCell>}
```
Thank you for this.

GeorgiSTodorov commented:
@pierrebeitz , the lint check is failing.

pierrebeitz commented:
thanks for having a look, @GeorgiSTodorov!

> Not related to this PR, but "with dismiss callback" doesn't seem to do anything.

i think there should be a message in the "actions"-panel when you click dismiss. is that not working?

GeorgiSTodorov commented:
> i think there should be a message in the "actions"-panel when you click dismiss. is that not working?

I don't remember if I checked there, I was expecting either an alert or a console.log.

mperrotti commented on packages/toaster/components/Toast.tsx:
```diff
@@ -26,21 +26,21 @@ export interface ToastProps {
 }
 
-const SuccessIcon = props => (
+const SuccessIcon = (
```
We could replace these with the Icon component. Toast components were created before we had the Icon component.

pierrebeitz replied:
i wanted to do that but storybook was not open at that moment - so i just compared the svg-paths which did not match up and abandoned the idea.

thanks a lot for paying close attention here! 😍

mperrotti commented on packages/toaster/components/Toast.tsx:
```diff
@@ -49,13 +49,13 @@ const SuccessIcon = props => (
 );
 
-const DangerIcon = props => (
+const DangerIcon = (
```
This could also be replaced using the Icon component

mperrotti commented on packages/toaster/components/Toast.tsx:
```diff
@@ -64,13 +64,13 @@ const DangerIcon = props => (
 );
 
-const WarningIcon = props => (
+const WarningIcon = (
```
This could also be replaced using the Icon component

mperrotti commented on packages/toaster/components/Toaster.tsx:
```diff
@@ -5,22 +5,21 @@ import { ToastProps, ToastId } from "./Toast";
+  ToasterProps,
+  { toasts: Array<React.ReactElement<ToastProps>> }
+> {
+  public timeouts: NodeJS.Timeout[] = [];
```
Nice. I didn't know about NodeJS.Timeout

mperrotti commented on typings/react.d.ts:
```diff
@@ -0,0 +1,7 @@
+  interface ReactChildren {
+    // we sometimes want to handle typed children, which react does not support out of the box.
+    // instead of jumping through hoops, we introduce this parameterized version of toArray.
+    toArray<A>(children?: A | A[]): A[];
```
Thanks! This will be useful

pierrebeitz commented:
@mperrotti addressed your feedback. i'd hit the button, but am not sure about what the conventions in ui-kit are. would it be ok to self-merge if there's only been small changes and like 3 people approved before?

mperrotti commented:
@pierrebeitz - it's ok to self-merge :)
